src/main/core/Subscriber/SecurityEventSubscriber.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Claroline\CoreBundle\Subscriber;
+
+use Claroline\CoreBundle\Entity\Log\LogSecurity;
+use Claroline\CoreBundle\Event\CatalogEvents\SecurityEvents;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Contracts\EventDispatcher\Event;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class SecurityEventSubscriber implements EventSubscriberInterface
+{
+    private $em;
+    private $client;
+    private $security;
+    private $requestStack;
+    private $translator;
+
+    public function __construct(
+        EntityManagerInterface $em,
+        HttpClientInterface $client,
+        Security $security,
+        RequestStack $requestStack,
+        TranslatorInterface $translator
+    ) {
+        $this->em = $em;
+        $this->client = $client;
+        $this->security = $security;
+        $this->requestStack = $requestStack;
+        $this->translator = $translator;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            SecurityEvents::USER_LOGIN => 'logEvent',
+            SecurityEvents::USER_LOGOUT => 'logEvent',
+            SecurityEvents::USER_DISABLE => 'logEvent',
+            SecurityEvents::USER_ENABLE => 'logEvent',
+            SecurityEvents::NEW_PASSWORD => 'logEvent',
+            SecurityEvents::FORGOT_PASSWORD => 'logEvent',
+            SecurityEvents::ADD_ROLE => 'logEvent',
+            SecurityEvents::REMOVE_ROLE => 'logEvent',
+            SecurityEvents::VIEW_AS => 'logEvent',
+        ];
+    }
+
+    public function logEvent(Event $event, string $eventName): void
+    {
+        $request = $this->requestStack->getCurrentRequest();
+
+        $logEntry = new LogSecurity();
+        $logEntry->setDetails(sprintf('Log %s: %s', $eventName, $event->getMessage($this->translator)));
+        $logEntry->setEvent($eventName);
+        $logEntry->setTarget($event->getUser());
+        $logEntry->setDoer($this->security->getUser());
+        $logEntry->setDoerIp($request->getClientIp());
+
+        //Get infos from ip address
+        $response = json_decode($this->client->request('GET', 'http://ip-api.com/json/'.$request->getClientIp()), true);
```

Comment:
<img width="716" alt="Screenshot 2021-02-26 at 11 32 54" src="https://user-images.githubusercontent.com/7502063/109289404-714e9d80-7826-11eb-804e-0ae9f7ffc5f5.png">

I think we have to catch `429` responses and set the geoloc-related properties to `null` in this case to avoid crashing.